mongodbatlas/third_party_integration.go

```diff
@@ -0,0 +1,203 @@
+package mongodbatlas
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+)
+
+const (
+	integrationBasePath = "groups/%s/integrations"
+)
+
+// IntegrationsService is an interface for interfacing with the Third-Party Integrations
+// endpoints of the MongoDB Atlas API.
+
+// See more: https://docs.atlas.mongodb.com/reference/api/third-party-integration-settings/
+type IntegrationsService interface {
+	Create(context.Context, string, string, *ThirdPartyService) (*IntegrationResponse, *Response, error)
+	Replace(context.Context, string, string, *ThirdPartyService) (*IntegrationResponse, *Response, error)
+	Delete(context.Context, string, string) (*Response, error)
+	Get(context.Context, string, string) (*ThirdPartyService, *Response, error)
+	List(context.Context, string) (*IntegrationResponse, *Response, error)
+}
+
+// TeamsServiceOp handles communication with the Teams related methods of the
+// MongoDB Atlas API
+type IntegrationsServiceOp service
+
+var _ IntegrationsService = &IntegrationsServiceOp{}
+
+// IntegrationRequest contains parameters for different third-party services
+type ThirdPartyService struct {
+	Type        string `json:"type,omitempty"`
+	LicenseKey  string `json:"licenseKey,omitempty"`
+	AccountID   string `json:"accountId,omitempty"`
+	WriteToken  string `json:"writeToken,omitempty"`
+	ReadToken   string `json:"readToken,omitempty"`
+	APIKey      string `json:"apiKey,omitempty"`
+	Region      string `json:"region,omitempty"`
+	ServiceKey  string `json:"serviceKey,omitempty"`
+	APIToken    string `json:"apiToken,omitempty"`
+	TeamName    string `json:"teamName,omitempty"`
+	ChannelName string `json:"channelName,omitempty"`
+	RoutingKey  string `json:"routingKey,omitempty"`
+	FlowName    string `json:"flowName,omitempty"`
+	OrgName     string `json:"orgName,omitempty"`
+	URL         string `json:"url,omitempty"`
+	Secret      string `json:"secret,omitempty"`
+}
+
+// IntegrationResponse contains the response from the endpoint
+type IntegrationResponse struct {
```

Comment:
prefer plural for collections
```suggestion
type IntegrationsResponse struct {
```